src/shared/install.c

```diff
@@ -849,6 +936,7 @@ static int install_info_add(
                 const char *path) {
         UnitFileInstallInfo *i = NULL;
         int r;
+        char *n;
```

Comment:
fail can be reached before n is assigned, so n should be initialized to NULL here.